fleet_adapter_mir_actions/fleet_adapter_mir_actions/rmf_wait_until.py

```diff
@@ -0,0 +1,485 @@
+                      'perform wait until action.'
+            )
+            return
+
+        self.update_gap = description.get(
+            'update_gap',
+            context.action_config.get('update_gap', 60))  # seconds
+        self.wait_timeout = description.get(
+            'timeout', context.action_config.get('timeout', 60))  # seconds
+        self.signal_config = context.action_config.get('signals')
+        self.default_signal = context.action_config.get('default_signal')
+
+        self.start_time = self.context.node.get_clock().now().nanoseconds / 1e9
+        self.context.node.get_logger().info(
+            f'New wait until action with a timeout of {self.wait_timeout} '
+            f'seconds requested for robot [{self.context.name}]')
+
+    def cancel_task(self, label: str = ''):
+        def _cancel_success():
+            pass
+
+        def _cancel_fail():
+            pass
+        self.cancel_task_of_action(_cancel_success, _cancel_fail, label)
+
+    def update_action(self):
+        # If the action is no longer active, mark action as complete
+        if self.execution is not None and not self.execution.okay():
+            self.context.node.get_logger().info(
+                f'[wait_until] action is no longer underway and valid, '
+                f'marking action as complete.')
+            return True
+
+        # Check if we've received a move off signal
+        if self.move_off_cb():
+            self.context.node.get_logger().info(
+                f'[{self.context.name}] has received move off signal, marking '
+                f'action as completed.'
+            )
+            self.execution.finished()
+            return True
+
+        # Check if the default timeout has passed
+        now = self.context.node.get_clock().now().nanoseconds / 1e9
+        if now > self.start_time + self.wait_timeout:
+            self.context.node.get_logger().info(
+                f'Robot [{self.context.name}] has completed waiting for '
+                f'{self.wait_timeout} seconds without move off signal, '
+                f'marking action as complete.'
+            )
+            return True
+
+        # Log the robot waiting every X seconds
+        seconds_passed = round(now - self.start_time)
+        if seconds_passed % self.update_gap == 0:
+            self.context.node.get_logger().info(
+                f'{seconds_passed} seconds have passed since robot '
+                f'[{self.context.name}] started its waiting action.'
+            )
+
+        return False
+
+    def create_move_off_cb(self, description: dict, custom_modules: dict):
+        signal_cb = None
+        signal_type = None
+
+        # Determine which move off signal config to use. Any config populated
+        # in the task description overrides the default config provided in
+        # action config.
+        if 'signal_type' in description:
+            signal_type = description['signal_type']
```

Comment:
The `signal_type` terminology is confusing here, because elsewhere in this package "signal type" refers to one of `mission`, `plc`, or `custom`, but here I believe it's meant to be one of the key names inside the `signals` dictionary. I think the resulting confusion may actually be the cause of some bugs later in this function.

I would suggest we call this `signal_name` instead of `signal_type`, including in the `description`.